editor/src/components/canvas/dom-walker.ts

```diff
@@ -349,21 +348,24 @@ function runSelectiveDomWalker(
           domWalkerMutableState.invalidatedPaths,
         )
 
-        workingMetadata.push(...collectedMetadata)
+        mergeMetadataMapsWithFragments_MUTATE(workingMetadata, collectedMetadata)
       }
     })
-    const rootMetadataForOtherElements = rootMetadataInStateRef.current.filter(
-      (m) => !elementsToFocusOn.some((p) => EP.pathsEqual(p, m.elementPath)),
+    const otherElementPaths = Object.keys(rootMetadataInStateRef.current).filter(
+      (path) => !elementsToFocusOn.some((focusPath) => EP.toString(focusPath) === path),
     )
+    const rootMetadataForOtherElements = pick(otherElementPaths, rootMetadataInStateRef.current)
+    mergeMetadataMapsWithFragments_MUTATE(rootMetadataForOtherElements, workingMetadata)
```

Comment:
I'm very uneasy about this, because we're very close to mutating the metadata stored in the editor state here. The only reason we're _not_ doing that right now is because the mutating functions above are replacing entire values, so please can you add comments to the mutating functions above making it explicitly clear that we should never mutate the metadata objects themselves.

Reply:
Oooh good idea! We could also capture this in a test case for the ultimate in developer protection

Reply:
I don't think a test is necessary, in this PR I modified `ElementInstanceMetadataMap` to contain readonly `ElementInstanceMetadata` values, so if I try to mutate them, I see a compile error. E.g.:

<img width="830" alt="image" src="https://user-images.githubusercontent.com/127662/173382024-66b09667-6bcf-4178-9d6d-0aeaeba6be11.png">

See this change:
<img width="1371" alt="image" src="https://user-images.githubusercontent.com/127662/173382398-825d4b06-9a83-41bd-8ce6-d52ecd41df62.png">

I'm happy to update the comment though.

Reply:
Added comments in 181c4649bef38075ce9f55c5b07a75c57caa8d7a

Reply:
Ah ok yep that works for me! 👍